src/mame/video/starfield_05xx.cpp

```diff
@@ -0,0 +1,683 @@
+#define LFSR_CYCLES_PER_LINE            256
+#define LFSR_HIT_MASK           0xFA14
+#define LFSR_HIT_VALUE          0x7800
+#define LFSR_SEED               0x7FFF
+
+
+static const int speed_X_cycle_count_offset[] =
+{
+	0,  1,  2,  3, -4, -3, -2, -1
+};
+
+static const int pre_vis_cycle_count_values[] =
+{
+	22 * LFSR_CYCLES_PER_LINE,
+	23 * LFSR_CYCLES_PER_LINE,
+	22 * LFSR_CYCLES_PER_LINE,
+	23 * LFSR_CYCLES_PER_LINE,
+	19 * LFSR_CYCLES_PER_LINE,
+	20 * LFSR_CYCLES_PER_LINE,
+	20 * LFSR_CYCLES_PER_LINE,
+	22 * LFSR_CYCLES_PER_LINE
+};
+
+static const int post_vis_cycle_count_values[] =
+{
+	10 * LFSR_CYCLES_PER_LINE,
+	10 * LFSR_CYCLES_PER_LINE,
+	12 * LFSR_CYCLES_PER_LINE,
+	12 * LFSR_CYCLES_PER_LINE,
+	9  * LFSR_CYCLES_PER_LINE,
+	9  * LFSR_CYCLES_PER_LINE,
+	10 * LFSR_CYCLES_PER_LINE,
+	9  * LFSR_CYCLES_PER_LINE
+};
+
+
+
+
+starfield_05xx_device::starfield_05xx_device(machine_config const &mconfig, char const *tag, device_t *owner, uint32_t clock)
+	: device_t(mconfig, STARFIELD_05XX, tag, owner, clock)
+	, m_enable(0)
+	, m_lfsr(LFSR_SEED)
+	, m_pre_vis_cycle_count(0)
+	, m_post_vis_cycle_count(0)
+	, m_starfield_set_a(0)
+	, m_starfield_set_b(0)
+	, m_starfield_offset_X(0)
+	, m_starfield_offset_Y(0)
+	, m_starfield_limit_X(0)
+
+{
+}
+
+
+
+void starfield_05xx_device::enable_starfield(uint8_t on)
+{
+	if (!on) m_lfsr = LFSR_SEED;
+
+	m_enable = on ? 1 : 0;
+}
+
+
+
+void starfield_05xx_device::set_scroll_speed(uint8_t index_X, uint8_t index_Y)
+{
+	// Set initial pre- and post- visible cycle counts based on vertical 
+	// scroll registers
+	m_pre_vis_cycle_count = pre_vis_cycle_count_values[index_Y];
+	m_post_vis_cycle_count = post_vis_cycle_count_values[index_Y];
+	
+	// X scrolling occurs during pre-visible portion, so adjust 
+	// pre-visible cycle count to based on horizontal scroll registers
+	m_pre_vis_cycle_count += speed_X_cycle_count_offset[index_X];
+}
+
+
+void starfield_05xx_device::set_active_starfield_sets(uint8_t set_a, uint8_t set_b)
+{
+	// Set active starfield sets based on starfield select registers
+	m_starfield_set_a = set_a;
+	m_starfield_set_b = set_b;
+}
+
+
+void starfield_05xx_device::set_starfield_config(uint16_t off_X, uint16_t off_Y, uint16_t lim_X)
+{
+	// Set X and Y starfield position offsets
+	m_starfield_offset_X = off_X;
+	m_starfield_offset_Y = off_Y;
+	
+	// Set X range limit
+	m_starfield_limit_X = lim_X;
+}
+
+
+uint16_t starfield_05xx_device::get_next_lfsr_state(uint16_t lfsr)
+{
+    uint16_t bit;
+
+    // 16-bit FIBONACCI-style LFSR with taps at 16,13,11, and 6
+    // These taps produce a maximal sequence of 65,535 steps.
+
+    bit = ((lfsr >> 0) ^ (lfsr >> 3) ^ (lfsr >> 5) ^ (lfsr >> 10));
+    lfsr = (lfsr >> 1) | (bit << 15);
+
+    return lfsr;
+}
+
+
+void starfield_05xx_device::draw_starfield(bitmap_ind16 &bitmap, const rectangle &cliprect, int flip)
+{
+	uint16_t pre_vis_cycle_count = m_pre_vis_cycle_count;
+	uint16_t post_vis_cycle_count = m_post_vis_cycle_count;
+
+	if ( m_enable )
```

Comment:
Please be consistent with spacing – in your other control statements you're not putting extra spaces around the condition like you have on this one.

Reply:
Change committed